plugin/semantic.py

```diff
@@ -0,0 +1,232 @@
+from .code_actions import CodeActionOrCommand
+from .core.protocol import Diagnostic
+from .core.protocol import Request
+from .core.registry import LspTextCommand
+from .core.registry import windows
+from .core.sessions import SessionBufferProtocol
+from .core.typing import List, Optional, Any, Dict, Tuple, Sequence, cast
+from .core.views import text_document_identifier
+
+import sublime
+import sublime_plugin
+import time
+
+SUBLIME_WORD_MASK = 515
+
+NON_MODIFIER_SCOPES = {
+            'variable'     :'variable.other.lsp',
+            'parameter'    :'variable.parameter.lsp',
+            'function'     :'variable.function.lsp',
+            'method'       :'variable.function.lsp',
+            'property'     :'variable.other.member.lsp',
+            'class'        :'entity.name.class.lsp',
+            'enum'         :'variable.enum.lsp',
+            'enumMember'   :'entity.name.enum.lsp',
+            'type'         :'storage.type.lsp',
+            'macro'        :'variable.other.constant.lsp',
```

Comment:
The (3rd-party) syntax for the Julia language (where macros are commonly used) uses `variable.macro`, which I think would fit better here. Not sure in which other languages something like macros are used and how they are scoped elsewhere.